core/src/main/java/com/graphhopper/util/shapes/GHPoint.java

```diff
@@ -35,28 +35,27 @@ public GHPoint(double lat, double lon) {
         this.lon = lon;
     }
 
-    public static GHPoint parse(String str) {
-        return parse(str, false);
+    public static GHPoint fromString(String str) {
+        return fromString(str, false);
     }
 
-    public static GHPoint parseLonLat(String str) {
-        return parse(str, true);
+    public static GHPoint fromStringLonLat(String str) {
+        return fromString(str, true);
     }
 
-    private static GHPoint parse(String str, boolean lonLatOrder) {
+    private static GHPoint fromString(String str, boolean lonLatOrder) {
         String[] fromStrs = str.split(",");
         if (fromStrs.length == 2) {
-            try {
-                double fromLat = Double.parseDouble(fromStrs[0]);
-                double fromLon = Double.parseDouble(fromStrs[1]);
-                if (lonLatOrder)
-                    return new GHPoint(fromLon, fromLat);
-
+            double fromLat = Double.parseDouble(fromStrs[0]);
+            double fromLon = Double.parseDouble(fromStrs[1]);
+            if (lonLatOrder) {
+                return new GHPoint(fromLon, fromLat);
+            } else {
                 return new GHPoint(fromLat, fromLon);
-            } catch (Exception ex) {
             }
+        } else {
+            throw new IllegalArgumentException(str);
```

Comment:
Maybe. I do such things incrementally and think, hey, an Exception is better than null (what was there before).